src/components/Dropdown.vue

```diff
@@ -0,0 +1,152 @@
+<script lang="ts">
+import { computed, defineComponent, PropType, ref } from "vue";
+import DropdownContent from "./DropdownContent.vue";
+
+export type Options = string[];
+
+type Props = {
+  options: Options;
+  selectedOption: string;
+  label: string;
+  width: string;
+};
+
+export default defineComponent({
+  components: { DropdownContent },
+  props: {
+    options: {
+      type: Object as PropType<Options>,
+      required: true,
+    },
+    selectedOption: {
+      type: String,
+      required: true,
+    },
+    label: {
+      type: String,
+      default: "",
+    },
+    width: {
+      type: String,
+      default: "8.8rem",
+    },
+  },
+  emits: ["update:selectedOption"],
+  setup(props: Props, { emit }) {
+    const isOptionsShown = ref(false);
```

Comment:
usetoggleがおすすめ
https://vueuse.org/shared/useToggle/
```suggestion
    const [isOptionsShown, toggleShown] = useToggle();
```

Reply:
こんなのがあったんですね。便利です！